backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/SchedulingManager.java

```diff
@@ -458,6 +461,30 @@ private PerHostMessages getDetails() {
         }
     }
 
+    private void updateNumaConsumptionOnHost(VDS host,
+                                             Map<Guid, Map<Integer, NumaNodeMemoryConsumption>> numaConsumptionPerVm) {
+        Map<Integer, NumaNodeMemoryConsumption> numaConsumption = numaConsumptionPerVm.values().stream()
+                .flatMap(m -> m.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, NumaNodeMemoryConsumption::merge));
+        updateHostNumaNodes(host, numaConsumption);
+    }
+
+    private void updateDedicatedNumaMemoryConsumption(VM vm, VDS host, String numaPinningString,
+                                                      Map<Guid, Map<Integer, NumaNodeMemoryConsumption>> numaConsumptionPerVm) {
+        if (vm.getCpuPinningPolicy() == CpuPinningPolicy.DEDICATED) {
+            Map<Integer, Collection<Integer>> currentNumaPinning = NumaPinningHelper.parseNumaMapping(numaPinningString);
+            for (VmNumaNode vmNumaNode : vm.getvNumaNodeList()) {
+                vmNumaNode.setVdsNumaNodeList(new ArrayList<>(currentNumaPinning.get(vmNumaNode.getIndex())));
```

Comment:
how about splitting this into:
var vdsNumaNodesPinned = currentNumaPinning.get(vmNumaNode.getIndex())
vmNumaNode.setVdsNumaNodeList(new ArrayList<>(vdsNumaNodesPinned));